.github/workflows/sonar.yaml

```diff
@@ -0,0 +1,23 @@
+name: Sonar Scan
+on: [push, pull_request_target]
+
+jobs:
+  sonar:
+    name: Sonar Scan
+    runs-on: ubuntu-latest
+    # always run on push events
+    # only run on pull_request_target event when pull request pulls from fork repository
+    if: >
+      github.event_name == 'push' || 
+      github.event_name == 'pull_request_target' && github.event.pull_request.head.repo.full_name != github.repository 
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
```

Comment:
Presume we need this to fetch all of the commits? Looks like default is 1

Reply:
Specifically, it's to make sure all the blame information is available to Sonar. Further information is here: https://community.sonarsource.com/t/sonarcloud-with-github-actions/22713